src/main/kotlin/no/nav/syfo/application/BlockingApplicationRunner.kt

```diff
@@ -161,20 +165,17 @@ class BlockingApplicationRunner {
 
                     val personNumberPatient = healthInformation.pasient.fodselsnummer.id
 
-                    val personNumberDoctor = if (!fnrOgDnrMangler(healthInformation)) {
-                        receiverBlock.avsenderFnrFraDigSignatur
-                    } else if (!hprManglerFraSignatur(fellesformat)) {
-                        val fnr = getFnrFromHpr(norskHelsenettClient, fellesformat, msgId)
-                        if (fnr.isNullOrEmpty()) {
-                            handleFnrAndDnrAndHprIsmissingFromBehandler(
-                                loggingMeta, fellesformat,
-                                ediLoggId, msgId, msgHead, env, kafkaproducerApprec, jedis, sha256String
-                            )
-                            continue@loop
-                        } else {
-                            fnr
-                        }
-                    } else {
+                    val signaturFnr: String = receiverBlock.avsenderFnrFraDigSignatur
+                    val behandlerFnr: String? = getFnrOrDnr(healthInformation)
+                    val behandlerHpr: String? = extractHpr(fellesformat)?.id
+
+                    val signerendeBehandler: Behandler? = getBehandler(signaturFnr, null, loggingMeta)
+                    val behandler: Behandler? = getBehandler(behandlerFnr, behandlerHpr, loggingMeta)
```

Comment:
Veldig ofte vil disse være like. Kanskje vi kan sjekke om de er like før vi slår dem opp to ganger? :)

Reply:
Haha, god idé! Vi snakket faktisk om å legge inn en sjekk her, men det forsvant nok i febertåka 😄